packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerFormConnected.js

```diff
@@ -11,12 +13,57 @@ class AddPrayerFormConnected extends React.Component {
 
   render() {
     return (
-      <Query query={getUserProfile} fetchPolicy={'cache-only'}>
-        {({ data }) => (
-          <AddPrayerForm
-            avatarSource={get(data, 'currentUser.profile.photo', { uri: null })}
-            {...this.props}
-          />
+      <Query query={GET_USER_PROFILE} fetchPolicy={'cache-only'}>
+        {(queryObj) => (
+          <Mutation
+            mutation={ADD_PRAYER}
+            update={(cache, { data: { addPublicPrayerRequest } }) => {
+              const { getCurrentPersonPrayerRequests } = cache.readQuery({
+                query: GET_PRAYERS,
+              });
+              cache.writeQuery({
+                query: GET_PRAYERS,
+                data: {
+                  getCurrentPersonPrayerRequests: getCurrentPersonPrayerRequests.concat(
+                    [addPublicPrayerRequest]
+                  ),
+                },
+              });
+            }}
+          >
+            {(addPrayer) => (
+              <AddPrayerForm
+                onSubmit={(values) => {
+                  addPrayer({
+                    variables: {
+                      campusID: get(
+                        queryObj.data,
```

Comment:
```suggestion
                       data,
```

Reply:
I don't like it either but I can't do that because another argument is defined in the scope below as `data`